AcadeMeData/conftest.py

```diff
@@ -9,3 +9,24 @@ def generate_university(university_id=5, name='The Technion', location="Haifa",
                             description=description)
     university.save()
     return university
+
+
+@pytest.fixture
+def generate_degree(degree_id=1, name='History', universities="Ben Gurion University, Reichman University",
+                    description="Learn about historic events and their influences on the world"):
+    degree = Degree.create_degree(degree_id=degree_id, name=name, universities=universities,
+                                  description=description)
+    degree.save()
+    return degree
+
+
+@pytest.fixture
+def generate_professor(generate_university, professor_id=2, name="DR Arnold Schwarzenegger",
+                       description="A cool guy who looked familiar", rate=4.5):
+    university = generate_university
+    professor = Professor.create_professor(professor_id=professor_id,

```

Comment:
Can you explain? I don't understand..
I've just copied the fixture from the merged 'professor_test.py' file.
Why do I need to pass the name?

Reply:
It is no a must, if you want to parse new university, since you are creating university with a name, if you parse unique name to the fixture
You can retrieve the university from proffessor object. and this coulkd be another assertion for instance, 
if all tests create the same University for test professor, you will need to do more checks to verify duplication and such

Reply:
I still don't quiet understand.. you want me to do something like this?
<img width="361" alt="Untitled" src="https://user-images.githubusercontent.com/89524327/146053715-78b1a30e-0c81-4ac7-b8c9-5ff5bd060b1f.png">

Reply:
@Yarboa @Omeramsc

Reply:
Look at line 26
university = generate_university you are calling a fixture from the test,
fixture is called in test declaration, so you are doing it in the declaration 
Please delete line 26.
Now you have an issue to find the university name,

And you are calling the generate_university but you need to parse the name you create to generate_university 

than, you can retrieve the University object by name as you pasted in the snippet is different than i wrote 
def generate_professor(generate_university, university_name
professor_id=2,...

Maybe you should try other teammates, maybe they understand

Reply:
@Yarboa 
I think I got it, fixed now!

Reply:
@Yarboa now it is fixed